content/en/docs/tasks/administer-cluster/kms-provider.md

```diff
@@ -195,6 +197,21 @@ To encrypt the data:
              timeout: 3s
    ```
 
+Setting `--encryption-provider-config-automatic-reload` to `true` collapses all health checks to a single health check endpoint. Individual health checks are only available when KMS v1 providers are in use and the encryption config is not auto-reloaded.
+
+Following table summarizes the health check endpoints for each KMS version:
+
+| KMS configurations        | Without Automatic Reload           | With Automatic Reload  |
+| ------------------------- |------------------------------------| -----------------------|
+| KMS v1 only               | Individual Healthchecks            | Single Healthcheck     |
+| KMS v2 only               | Single Healthcheck                 | Single Healthcheck     |
+| Both KMS v1 and v2        | Individual Healthchecks            | Single Healthcheck     |
+| No KMS                    | None                               | Single Healthcheck     |
+
+`Single Healthcheck` means that the only health check endpoint is `/healthz/kms-providers`.
+
+`Individual Healthchecks` means that each KMS plugin has an associated health check endpoint based on its location in the encryption config: `/healthz/kms-providers-0`, `/healthz/kms-providers-1` etc.
```

Comment:
Can you clarify are these endpoints derived or are these examples? If they are configurable, who is responsible for configuring these healthcheck endpoints and where?

Reply:
The endpoint paths are hard coded and generated/controlled by the server, so they are always:

`/healthz/kms-provider-0`
`/healthz/kms-provider-1`

Where the index corresponds to the order in which the KMS config is processed.

Reply:
> The endpoint paths are hard coded and generated/controlled by the server, so they are always:
> 
> `/healthz/kms-provider-0` `/healthz/kms-provider-1` `...` `/healthz/kms-provider-N`
> 
> Where the index corresponds to the order in which the KMS config is processed.

Would be helpful to add that so people know its not something they have to configure and it's system generated.

Reply:
added a note.